### ISSUE TITLE
Improve open-vsx.org support

### ISSUE THREAD

**Issue by berzoidberg** [EclipseFdn/open-vsx.org#659]
On August 3rd, a bug was filed for an issue with the open-vsx.org deployment of openvsx indicating that CORS headers were not properly flowing. Due to security changes in more recent versions of vscode, this meant the site (and thusly, built-in extension management using open-vsx.org) was inoperable for many users.

As of today, this issue is resolved via an NGINX configuration change.

I'm opening this bug to discuss methods of improving the response time to these types of issues. As per @eclipsewebmaster, Eclipse currently provides hosting of open-vsx but does not currently employ developers to maintain the codebase or seemingly to dedicated time to the maintenance of this deployment.

As popularity of open-vsx.org increases through the adoption of pure-OSS distributions of vscode, open-vsx.org is likely to come under more strain and impact more users when issues arise. Accordingly, we need to as a community find a way to increase the response time to these issues.

Since this issue is filed under the open-vsx.org project, which pertains directly to the hosting of the application -- let's discuss what we can do to improve response time for infrastructure-related issues. This includes NGINX configuration, database backups and outages, failovers, etc etc.

To the Eclipse team: What can be done here? Is Eclipse willing to back this publicly hosted instance with a proper SLA and devote developer time to solving these issue in a timely manner? Alternatively, is Eclipse willing to donate hosting resources to a volunteer team of engineers who volunteer to response to these issues promptly?

**Comment by eclipsewebmaster** [comment]
Thanks for filing this. We'll have a response for you shortly.

**Comment by eclipsewebmaster** [comment]
Thanks for your patience. We're working on a postmortem and recommendations.

**Comment by brianking** [comment]
Thank you for opening this issue and your commitment to Open VSX.

**tl;dr**
While the Eclipse Foundation hosts the OpenVSX Registry service, the Eclipse Foundation relies entirely on the open source community and funding from the Cloud DevTools working group for the development of the codebase that powers open-vsx.org. After early investigations into the CORS issue, we believed it was a server code issue that the Foundation staff didn’t have the expertise to fix. After receiving more urging, and with help from the community, we were eventually able to find a workaround that resolved the issue. However more work needs to be done to improve how CORS requests are handled in the server code.

**More Details**
The Open VSX Registry service (code at [https://github.com/EclipseFdn/open-vsx.org](https://github.com/EclipseFdn/open-vsx.org)) is based on the Open VSX project (code at [https://github.com/eclipse/openvsx](https://github.com/eclipse/openvsx)). The Eclipse Foundation is responsible for the running of the service at [https://open-vsx.org/](https://open-vsx.org), which includes hosting and infrastructure. The Open VSX Registry is not covered by SLA because it is not deemed to be a production service, the service is based on project code at [version 0.1.0](https://github.com/eclipse/openvsx/releases) at the time of writing.

The Open VSX project is an independent open source project. While we aim to work closely with the project to fix issues and feed other input to them that might drive development, there has been some disconnect recently. The primary reason for this is simply that the project has been struggling to get contributions.

We encourage contributions and new committers to the project, whether that be from companies / projects that are integrating Open VSX into their software or from individuals. That is companies that depend on the Open VSX Registry service to distribute their VSCode extensions and companies that host their own internal registry based on Eclipse Open VSX project. To support these contributions, Committers may be given access to a staging site - this is evaluated on a case-by-case basis. One particular challenge is that it is difficult for us to give access to our infrastructure for testing and development because of GDPR and security issues. We also encourage donations to the project, which are enabled by the [Working Group Sponsorship Agreement](https://www.eclipse.org/org/workinggroups/sponsorship/working-group-sponsorship-agreement.pdf).

In terms of the foundation donating hosting resources for other instances, this would need to be evaluated on a case-by-case basis by the Cloud DevTools Working Group, which is an Eclipse Foundation Working Group that supports the project. Regardless of where resources come from, we encourage integrators, which produce a high volume of traffic, to set up their own instances of Open VSX where they can host their own extensions and transparently proxy to  open-vsx.org as needed. This would have the effect of lessening the load on the service. Documentation for doing this can currently be found at:
[https://github.com/eclipse/openvsx/wiki/Deploying-Open-VSX](https://github.com/eclipse/openvsx/wiki/Deploying-Open-VSX)
[https://github.com/eclipse/openvsx#oauth-setup](https://github.com/eclipse/openvsx#oauth-setup)

We will strive to work as closely as possible with contributors who work in the open on the project. This includes sharing configurations, and listening to recommendations and adopting improvements made. As was shown during the recent resolution to the CORS issue, working together is powerful and makes the project and service successful. We are committed to continuing to host the Open VSX Registry service and are encouraged by the increasing number of integrations, listings in the registry, and the passion of the community.

**Comment by brianking** [comment]
Support for the site and further development have improved things in recent months. Closing this. If there is anything specific that needs to be addressed, please open a new issue.
